prow/plugins/cherrypickunapproved/cherrypick-unapproved.go

```diff
@@ -81,26 +82,57 @@ func handlePullRequest(pc plugins.Agent, pr github.PullRequestEvent) error {
 }
 
 func handlePR(gc githubClient, log *logrus.Entry, pr *github.PullRequestEvent, cp commentPruner, branchRe *regexp.Regexp, commentBody string) error {
-	// Only consider the events that indicate opening of the PR and
-	// when the cpApproved and cpUnapproved labels are added or removed
-	cpLabelUpdated := (pr.Action == github.PullRequestActionLabeled || pr.Action == github.PullRequestActionUnlabeled) &&
-		(pr.Label.Name == labels.CpApproved || pr.Label.Name == labels.CpUnapproved)
-	if pr.Action != github.PullRequestActionOpened && pr.Action != github.PullRequestActionReopened && !cpLabelUpdated {
-		return nil
-	}
-
 	var (
 		org    = pr.Repo.Owner.Login
 		repo   = pr.Repo.Name
 		branch = pr.PullRequest.Base.Ref
 	)
 
-	// if the branch doesn't match against the branch names allowed for cherry-picks,
-	// don't do anything
-	if !branchRe.MatchString(branch) {
-		return nil
+	switch pr.Action {
+	case github.PullRequestActionOpened, github.PullRequestActionReopened:
+		if !branchRe.MatchString(branch) {
+			return nil
+		}
+		return ensureLabels(gc, org, repo, pr, log, cp, commentBody)
+	case github.PullRequestActionLabeled, github.PullRequestActionUnlabeled:
+		if !branchRe.MatchString(branch) {
+			return nil
+		}
+		if !(pr.Label.Name == labels.CpApproved || pr.Label.Name == labels.CpUnapproved) {
+			return nil
+		}
+		return ensureLabels(gc, org, repo, pr, log, cp, commentBody)
+	case github.PullRequestActionEdited:
+		// if someone changes the base of their PR, we will get this event
+		// and the changes field will list that the base SHA and ref changes
+		var changes struct {
+			Base struct {
+				Ref struct {
+					From string `json:"from"`
+				} `json:"ref"`
+				Sha struct {
+					From string `json:"from"`
+				} `json:"sha"`
+			} `json:"base"`
+		}
+		if err := json.Unmarshal(pr.Changes, &changes); err != nil {
```

Comment:
So if something else than the base branch was changed (say the title) the deserialization here will fail? (The github docs are not really clear on this)

Reply:
It turns out the deserialization won't fail, but rather unmarshal everything to empty values.

I'll add a check to make sure that `changes.Base.Ref.From` is not  `""`